src/main/java/org/team1540/robot2025/subsystems/arm/ArmIOReal.java

```diff
@@ -0,0 +1,134 @@
+package org.team1540.robot2025.subsystems.arm;
+
+import static org.team1540.robot2025.subsystems.arm.ArmConstants.*;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.*;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.units.measure.*;
+
+public class ArmIOReal implements ArmIO {
+    private final TalonFX motor = new TalonFX(MOTOR_ID);
+    private final CANcoder cancoder = new CANcoder(CANCODER_ID);
+
+    private final StatusSignal<Angle> position = motor.getPosition();
+    private final StatusSignal<AngularVelocity> velocity = motor.getVelocity();
+    private final StatusSignal<Voltage> appliedVoltage = motor.getMotorVoltage();
+    private final StatusSignal<Current> current = motor.getSupplyCurrent();
+    private final StatusSignal<Temperature> temp = motor.getDeviceTemp();
+    private final StatusSignal<ForwardLimitValue> forwardLimit = motor.getForwardLimit();
+    private final StatusSignal<ReverseLimitValue> reverseLimit = motor.getReverseLimit();
+
+    private final MotionMagicVoltage positionCtrlReq = new MotionMagicVoltage(0).withSlot(0);
+    private final VoltageOut voltageCtrlReq = new VoltageOut(0);
+
+    // constructor
+    public ArmIOReal() {
+        TalonFXConfiguration motorConfig = new TalonFXConfiguration();
+
+        motorConfig.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+        motorConfig.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+
+        motorConfig.Feedback.FeedbackSensorSource = FeedbackSensorSourceValue.FusedCANcoder;
+        motorConfig.Feedback.FeedbackRemoteSensorID = CANCODER_ID;
+        motorConfig.Feedback.SensorToMechanismRatio = CANCODER_TO_PIVOT_RATIO * MOTOR_TO_CANCODER;
+        motorConfig.Feedback.RotorToSensorRatio = 1;
```

Comment:
This would be the same as MOTOR_TO_CANCODER. It is 1 (for now) but would be clearer to use the constant imo